sky/compositor/paint_context.h

```diff
@@ -13,21 +15,52 @@ namespace compositor {
 
 class PaintContext {
  public:
-  PaintContext(PictureRasterzier& rasterizer,
-               GrContext* gr_context,
-               SkCanvas* canvas);
+  class ScopedFrame {
+   public:
+    PaintContext& paint_context() { return context_; };
+
+    GrContext* gr_context() { return gr_context_; }
+
+    SkCanvas& canvas() { return canvas_; }
+
+    ScopedFrame(ScopedFrame&& frame) = default;
+
+    ~ScopedFrame() { context_.endFrame(); }
+
+   private:
+    PaintContext& context_;
+    SkCanvas& canvas_;
+    GrContext* gr_context_;
+
+    ScopedFrame() = delete;
```

Comment:
Is this necessary?  I thought the compiler did this for you when you declared a constructor with arguments.